### PR TITLE
chore(upgrade): update yml

### PR DESCRIPTION
**Pull request by xzf0587**

### CODE REVIEW

File: packages/fx-core/tests/core/middleware/testAssets/v3Migration/debug/transparent-m365-me/expected/app.local.yml [hunk 1]
```diff
@@ -43,9 +43,9 @@ provision:
   - uses: m365Title/acquire # Upload your app to Outlook and the Microsoft 365 app
     with:
       appPackagePath: ./build/appPackage/appPackage.${{TEAMSFX_ENV}}.zip # Relative path to the built app package.
-    # Output: following environment variable will be persisted in current environment's .env file.
-    # M365_TITLE_ID: the id of M365 title
-    # M365_APP_ID: the app id of M365 title
+    writeToEnvironmentFile:
+      titleId: M365_TITLE_ID
+      appId: M365_APP_ID
```
Comment (xzf0587) [inline packages/fx-core/tests/core/middleware/testAssets/v3Migration/debug/transparent-m365-me/expected/app.local.yml:48]: @kuojianlu please help review m365Title/acquire. There is no comment for it yet.

File: packages/fx-core/tests/core/middleware/testAssets/v3Migration/debug/transparent-tab-bot-func/expected/app.local.yml [hunk 2]
```diff
@@ -78,9 +78,11 @@ deploy:
...
       func: true
       dotnet: true
-    # Output: following environment variable will be persisted in current environment's .env file.
-    # SSL_CRT_FILE: certificate file
-    # SSL_KEY_FILE: certificate key
+    writeToEnvironmentFile: # Write the information of installed dependencies into environment file for the specified environment variable(s).
+      sslCertFile: SSL_CRT_FILE
+      sslKeyFile: SSL_KEY_FILE
+      funcPath: FUNC_PATH
+      dotnetPath: DOTNET_PATH
```
Comment (xzf0587) [inline packages/fx-core/tests/core/middleware/testAssets/v3Migration/debug/transparent-tab-bot-func/expected/app.local.yml:85]: @kuojianlu please help review prerequisite/install

File: packages/fx-core/tests/core/middleware/testAssets/v3Migration/debug/transparent-tab-bot-func/expected/app.local.yml [hunk 3]
```diff
@@ -1,30 +1,30 @@
...
 
   - uses: botAadApp/create # Creates a new AAD app for bot if BOT_ID environment variable is empty
     with:
       name: ${{CONFIG__MANIFEST__APPNAME__SHORT}}-bot
-    # Output: following environment variable will be persisted in current environment's .env file.
-    # BOT_ID: the AAD app client id created for bot
-    # SECRET_BOT_PASSWORD: the AAD app client secret created for bot
+    writeToEnvironmentFile:
+      botId: BOT_ID
+      botPassword: SECRET_BOT_PASSWORD
```
Comment (xzf0587) [inline packages/fx-core/tests/core/middleware/testAssets/v3Migration/debug/transparent-tab-bot-func/expected/app.local.yml:27]: @IvanJobs please help review botAadApp/create

File: packages/fx-core/tests/core/middleware/testAssets/v3Migration/jsSsoTab/expectedResult/js.app.yml [hunk 4]
```diff
@@ -5,22 +5,22 @@ projectId: 4d4bfff9-82bf-4d75-8e66-72982c48d1db
...
+      tenantId: AAD_APP_TENANT_ID
+      authority: AAD_APP_OAUTH_AUTHORITY
+      authorityHost: AAD_APP_OAUTH_AUTHORITY_HOST
   - uses: teamsApp/create # Creates a Teams app
     with:
       name: ${{CONFIG__MANIFEST__APPNAME__SHORT}} # Teams app name
-    # Output: following environment variable will be persisted in current environment's .env file.
-    # TEAMS_APP_ID: the id of Teams app
+    writeToEnvironmentFile:
+      teamsAppId: TEAMS_APP_ID
```
Comment (xzf0587) [inline packages/fx-core/tests/core/middleware/testAssets/v3Migration/jsSsoTab/expectedResult/js.app.yml:23]: @nliu-ms please help review teamsApp/create

File: packages/fx-core/tests/core/middleware/testAssets/v3Migration/jsSsoTab/expectedResult/js.app.yml [hunk 5]
```diff
@@ -54,8 +54,8 @@ provision:
   - uses: teamsApp/update # Apply the Teams app manifest to an existing Teams app. Will use the app id in manifest file to determine which Teams app to update.
     with:
       appPackagePath: ./build/appPackage/appPackage.${{TEAMSFX_ENV}}.zip # Relative path to this file. This is the path for built zip file.
-    # Output: following environment variable will be persisted in current environment's .env file.
-    # TEAMS_APP_ID: the id of Teams app
+    writeToEnvironmentFile:
+      teamsAppId: TEAMS_APP_ID
```
Comment (xzf0587) [inline packages/fx-core/tests/core/middleware/testAssets/v3Migration/jsSsoTab/expectedResult/js.app.yml:58]: @nliu-ms please help review teamsApp/update

File: packages/fx-core/tests/core/middleware/testAssets/v3Migration/jsSsoTab/expectedResult/js.app.yml [hunk 6]
```diff
@@ -95,5 +95,5 @@ publish:
   - uses: teamsApp/publishAppPackage # Publish the app to Teams Admin Center (https://admin.teams.microsoft.com/policies/manage-apps) for review and approval
     with:
       appPackagePath: ./build/appPackage/appPackage.${{TEAMSFX_ENV}}.zip
-  # Output: following environment variable will be persisted in current environment's .env file.
-  # TEAMS_APP_PUBLISHED_APP_ID: app id in Teams tenant app catalog.
\ No newline at end of file
+    writeToEnvironmentFile:
+      publishedAppId: PUBLISHED_APP_ID
```
Comment (xzf0587) [inline packages/fx-core/tests/core/middleware/testAssets/v3Migration/jsSsoTab/expectedResult/js.app.yml:99]: @nliu-ms please help review teamsApp/publishAppPackage. The comment has been removed.

File: packages/fx-core/tests/core/middleware/testAssets/v3Migration/debug/transparent-m365-me/expected/app.local.yml (hunk 1 above)
Comment (kuojianlu) [reply packages/fx-core/tests/core/middleware/testAssets/v3Migration/debug/transparent-m365-me/expected/app.local.yml:48]: I think we can add commet for it.

File: packages/fx-core/tests/core/middleware/testAssets/v3Migration/jsSsoTab/expectedResult/js.app.yml (hunk 6 above)
Comment (xzf0587) [reply packages/fx-core/tests/core/middleware/testAssets/v3Migration/jsSsoTab/expectedResult/js.app.yml]: Update it. Thanks.

File: packages/fx-core/tests/core/middleware/testAssets/v3Migration/debug/transparent-m365-me/expected/app.local.yml (hunk 1 above)
Comment (xzf0587) [reply packages/fx-core/tests/core/middleware/testAssets/v3Migration/debug/transparent-m365-me/expected/app.local.yml:48]: Updated it. Thanks.

File: packages/fx-core/templates/core/v3Migration/js.ts.app.local.yml [hunk 7]
```diff
@@ -124,11 +124,17 @@ deploy:
...
+    writeToEnvironmentFile: # Write the information of installed dependencies into environment file for the specified environment variable(s).
+      {{#devCert}}
+      sslCertFile: SSL_CRT_FILE
+      sslKeyFile: SSL_KEY_FILE
+      {{/devCert}}
+      {{#func}}
+      funcPath: FUNC_PATH
+      {{/func}}
+      {{#dotnet}}
+      dotnetPath: DOTNET_PATH
```
Comment (blackchoey) [inline packages/fx-core/templates/core/v3Migration/js.ts.app.local.yml:136]: The `cli/runDotnetCommand` sets `execPath: {{execPath}}` (see line 146), should the env var name also be set to `execPath`? The goal is avoid mismatch when the logic changes one place but forgets another place.

Comment (blackchoey) [reply packages/fx-core/templates/core/v3Migration/js.ts.app.local.yml:136]: Also need to see whether FUNC_PATH needs to do same thing

File: packages/fx-core/templates/core/v3Migration/js.ts.app.yml [hunk 8]
```diff
@@ -152,6 +152,8 @@ deploy:
   - uses: prerequisite/install # Install dependencies
     with:      
       dotnet: true
+    writeToEnvironmentFile: # Write the information of installed dependencies into environment file for the specified environment variable(s).
+      dotnetPath: DOTNET_PATH
```
Comment (blackchoey) [inline packages/fx-core/templates/core/v3Migration/js.ts.app.yml:156]: Also check whether needs to align with `dotnetPath` in line 166

File: packages/fx-core/templates/core/v3Migration/js.ts.app.local.yml (hunk 7 above)
Comment (xzf0587) [reply packages/fx-core/templates/core/v3Migration/js.ts.app.local.yml:136]: @kuojianlu  please help check it. The output in prerequisite/install and the input in cli/runDotnetCommand

Comment (xzf0587) [reply packages/fx-core/templates/core/v3Migration/js.ts.app.local.yml:136]: The {{execPath}} in line 146 will be replaced with DOTNET_PATH by taskMigration migrateValidateLocalPrerequisites.
So the expected output will be  execPath: ${{DOTNET_PATH}}

File: packages/fx-core/templates/core/v3Migration/js.ts.app.yml (hunk 8 above)
Comment (xzf0587) [reply packages/fx-core/templates/core/v3Migration/js.ts.app.yml:156]: same as above.

**Pull request merged**